ext/geometry/01_geometry.js

```diff
@@ -0,0 +1,2090 @@
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M11];
+  }
+  get m12() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M12];
+  }
+  get m13() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M13];
+  }
+  get m14() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M14];
+  }
+  get m21() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M21];
+  }
+  get m22() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M22];
+  }
+  get m23() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M23];
+  }
+  get m24() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M24];
+  }
+  get m31() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M31];
+  }
+  get m32() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M32];
+  }
+  get m33() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M33];
+  }
+  get m34() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M34];
+  }
+  get m41() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M41];
+  }
+  get m42() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M42];
+  }
+  get m43() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M43];
+  }
+  get m44() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_raw][INDEX_M44];
+  }
+  get is2D() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return this[_is2D];
+  }
+  get isIdentity() {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    return isIdentityMatrix(this);
+  }
+
+  translate(tx = 0, ty = 0, tz = 0) {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    const matrix = webidl.createBranded(DOMMatrix);
+    matrix[_writable] = true;
+    matrix[_raw] = new Float64Array(this[_raw]);
+    op_geometry_translate_self(
+      webidl.converters["unrestricted double"](tx),
+      webidl.converters["unrestricted double"](ty),
+      webidl.converters["unrestricted double"](tz),
+      matrix[_raw],
+    );
+    matrix[_is2D] = this[_is2D] && tz === 0;
+    return matrix;
+  }
+
+  scale(
+    scaleX = 1,
+    scaleY = scaleX,
+    scaleZ = 1,
+    originX = 0,
+    originY = 0,
+    originZ = 0,
+  ) {
+    webidl.assertBranded(this, DOMMatrixReadOnlyPrototype);
+    originX = webidl.converters["unrestricted double"](originX);
+    originY = webidl.converters["unrestricted double"](originY);
+    originZ = webidl.converters["unrestricted double"](originZ);
```

Comment:
`scaleZ` should be converted before `originX`, and `scaleX` should be converted only once.

```js
const init = {
  m11:  1, m21:  2, m31:  3, m41:  4,
  m12:  5, m22:  6, m32:  7, m42:  8,
  m13:  9, m23: 10, m33: 11, m43: 12,
  m14: 13, m24: 14, m34: 15, m44: 16,
};
const expected = {
  m11:  2, m21:  4, m31:  9, m41:  -46,
  m12: 10, m22: 12, m32: 21, m42: -126,
  m13: 18, m23: 20, m33: 33, m43: -206,
  m14: 26, m24: 28, m34: 45, m44: -286,
};

const obj = {
  next: 2,
  valueOf() {
    return this.next++;
  },
};
const m1 = DOMMatrix.fromMatrix(init).scale(obj, undefined, obj, obj, obj, obj);
const m2 = DOMMatrix.fromMatrix(init).scale(2, 2, 3, 4, 5, 6);
const m3 = DOMMatrix.fromMatrix(expected);
```

After evaluating the code above, the matrices represented by `m1`, `m2` and `m3` should be equal.